Assets/RGLUnityPlugin/Scripts/RadarSensor.cs

```diff
@@ -0,0 +1,192 @@
+        public RadarConfiguration configuration = RadarConfigurationLibrary.SmartmicroDRVEGRD169MediumRange;
+
+        private RGLNodeSequence rglGraphRadar;
+        private RGLNodeSequence rglSubgraphToWorldFrame;
+        private SceneManager sceneManager;
+
+        private const string RadarRaysNodeId = "RADAR_RAYS";
+        private const string RadarRangeNodeId = "RADAR_RANGE";
+        private const string RadarPoseNodeId = "RADAR_POSE";
+        private const string RadarRaytraceNodeId = "RADAR_RAYTRACE";
+        private const string CompactNodeId = "COMPACT";
+        private const string ToRadarFrameId = "TO_RADAR_FRAME";
+        private const string ToWorldFrameId = "TO_WORLD_FRAME";
+        private const string RemoveGroundNodeId = "REMOVE_GROUND";
+        private const string RadarPostprocessNodeId = "RADAR_POSTPROCESS";
+        private const string NoiseDistanceNodeId = "NOISE_DISTANCE";
+        private const string NoiseRaysNodeId = "NOISE_RAYS";
+
+        private RadarModel? validatedPreset = null;
+
+        private float timer;
+        private int fixedUpdatesInCurrentFrame = 0;
+        private int lastUpdateFrame = -1;
+
+        // Remove Ground Node parameters
+        private const float GroundAngleThreshold = 5.0f;
+        private const float GroundDistanceThreshold = 0.005f;
+        private const float GroundFilterDistance = 0.5f;
+
+        public void Awake()
+        {
+            rglGraphRadar = new RGLNodeSequence()
+                .AddNodeRaysFromMat3x4f(RadarRaysNodeId, new Matrix4x4[1] { Matrix4x4.identity })
+                .AddNodeRaysSetRange(RadarRangeNodeId, new Vector2[1] { new Vector2(0.0f, Mathf.Infinity) })
+                .AddNodeRaysTransform(RadarPoseNodeId, Matrix4x4.identity)
+                .AddNodeGaussianNoiseAngularRay(NoiseRaysNodeId, 0, 0)
+                .AddNodeRaytrace(RadarRaytraceNodeId)
+                .AddNodePointsCompact(CompactNodeId)
+                .AddNodeGaussianNoiseDistance(NoiseDistanceNodeId, 0, 0, 0)
+                .AddNodePointsTransform(ToRadarFrameId, Matrix4x4.identity)
+                .AddNodePointsRemoveGround(RemoveGroundNodeId, GroundAngleThreshold * Mathf.Deg2Rad, GroundDistanceThreshold, GroundFilterDistance)
+                .AddNodePointsRadarPostprocess(RadarPostprocessNodeId, 0.1f, 0.1f);
+
+            rglSubgraphToWorldFrame = new RGLNodeSequence()
+                .AddNodePointsTransform(ToWorldFrameId, Matrix4x4.identity);
+
+            RGLNodeSequence.Connect(rglGraphRadar, rglSubgraphToWorldFrame);
+        }
+
+        public void Start()
+        {
+            sceneManager = FindObjectOfType<SceneManager>();
+            if (sceneManager == null)
+            {
+                // TODO(prybicki): this is too tedious, implement automatic instantiation of RGL Scene Manager
+                Debug.LogError($"RGL Scene Manager is not present on the scene. Destroying {name}.");
+                Destroy(this);
```

Comment:
We may consider returning after this Destroy - in some cases this may cause problems, when we would like to call OnValidate below. In best case scenario, OnValidate may be called unnecessary.